lib/boolean.js

```diff
@@ -23,23 +28,72 @@ internals.Boolean = class extends Any {
             value
         };
 
-        if (typeof value === 'string' &&
-            options.convert) {
+        result.value = (this._inner._truthySet.has(value) ? true
+            : (this._inner._falsySet.has(value) ? false : value));
+
+        result.errors = (typeof result.value === 'boolean') ? null : this.createError('boolean.base', null, state, options);
+        return result;
+    }
+
+    _truthy() {
+
+        const values = Hoek.flatten(Array.prototype.slice.call(arguments));
+        for (let i = 0; i < values.length; ++i) {
+            const value = values[i];
+
+            Hoek.assert(value !== undefined, 'Cannot call truthy/falsy with undefined');
+            this._inner._truthySet.add(value);
+        }
+    }
+
+    truthy() {
+
+        const obj = this.clone();
+        obj._truthy.apply(obj, arguments);
+        return obj;
+    }
+
+    _falsy() {
+
+        const values = Hoek.flatten(Array.prototype.slice.call(arguments));
+        for (let i = 0; i < values.length; ++i) {
+            const value = values[i];
 
-            const lower = value.toLowerCase();
-            result.value = (lower === 'true' || lower === 'yes' || lower === 'on' || lower === '1' ? true
-                                                            : (lower === 'false' || lower === 'no' || lower === 'off' || lower === '0' ? false : value));
+            Hoek.assert(value !== undefined, 'Cannot call truthy/falsy with undefined');
+            this._inner._falsySet.add(value);
         }
+    }
+
+    falsy() {
+
+        const obj = this.clone();
+        obj._falsy.apply(obj, arguments);
+        return obj;
+    }
+
+    describe() {
 
-        if (typeof value === 'number' &&
-            options.convert) {
+        const description = Any.prototype.describe.call(this);
 
-            result.value = (value === 1 ? true
-                    : (value === 0 ? false : value));
+        if (this._inner._truthySet.values().length) {
```

Comment:
values() is already a copy, so I guess `description.truthyValues = this._inner._truthySet.values()` is enough.